### PR TITLE
PYIC-7072: inherited identities

### PR DESCRIPTION
**Pull request by sam803**
### What changed

Add tests for inherited idenities

### Why did it change

To increase API test coverage

### Issue tracking

https://govukverify.atlassian.net/browse/PYIC-7072

### Environment variables or secrets

- [x] No environment variables or secrets were added or changed

### CODE REVIEW

File: api-tests/src/utils/ipv-session.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export class IPVSessionDetails {
```
Comment (Joe-Edwards-GDS) [inline api-tests/src/utils/ipv-session.ts]: I generally try and avoid using `class` in Typescript unless I've got a really good reason...

Can this just be a plain old JS object?
```
interface IpvSessionDetails {
  subject: string;
  journeyId: string;
  journeyType: string;
  isReproveIdentity: boolean;
  inheritedIdentityId?: string;
}
```

File: api-tests/src/steps/ipv-steps.ts [hunk 2]
```diff
@@ -58,20 +60,24 @@ After(function (this: World, options: ITestCaseHookParameter) {
 });
 
 When(
-  /I start a new ?'([\w-]+)' journey( with reprove identity)?/,
+  /I start a new ?'([\w-]+)' journey( with reprove identity)?(:? with inherited identity '([\w-]+)')?/,
```
Comment (Joe-Edwards-GDS) [inline api-tests/src/steps/ipv-steps.ts]: What's the `:?` for?

File: api-tests/src/steps/ipv-steps.ts [hunk 3]
```diff
@@ -58,20 +60,24 @@ After(function (this: World, options: ITestCaseHookParameter) {
...
   async function (
     this: World,
     journeyType: string,
     reproveIdentity: string,
+    inheritedIdentityId: string,
```
Comment (Joe-Edwards-GDS) [inline api-tests/src/steps/ipv-steps.ts]: these last two should be `string | undefined`, I think?

File: api-tests/data/jar-requests/medium-confidence-pcl200-pcl250.json
Comment (Joe-Edwards-GDS) [inline api-tests/data/jar-requests/medium-confidence-pcl200-pcl250.json:1]: I'm slightly wondering if it's valuable storing these as separate JSON files - I suppose it makes it super-clear, but there is a lot of duplication when they only vary in the VTR field...

What do you think?

File: api-tests/features/inherited-identity.feature [hunk 4]
```diff
@@ -0,0 +1,36 @@
+Feature: Inherited Identity
```
Comment (Joe-Edwards-GDS) [inline api-tests/features/inherited-identity.feature:2]: Might want to tag the Feature (not the scenarios) with `@Build` - assuming they work against build?

**Review by Joe-Edwards-GDS — COMMENTED** [review]
Sorry I've given you a merge conflict!

File: api-tests/src/steps/ipv-steps.ts (hunk 2 above)
Comment (sam803) [reply api-tests/src/steps/ipv-steps.ts]: A mistake - I intended for it to be a non-capturing group `?:`

File: api-tests/data/jar-requests/medium-confidence-pcl200-pcl250.json
Comment (sam803) [reply api-tests/data/jar-requests/medium-confidence-pcl200-pcl250.json:1]: Yes I did consider passing the vtr to `generateJar` function but would add more complexity to the [regex](https://github.com/govuk-one-login/ipv-core-back/blob/163602bc0840efcc3f481c42c87554da742ac960/api-tests/src/steps/ipv-steps.ts#L62) in the step so I opted for this approach - happy to change though :)

File: api-tests/src/utils/ipv-session.ts (hunk 1 above)
Comment (sam803) [reply api-tests/src/utils/ipv-session.ts]: Yes, sure (typescript novice here!)

File: api-tests/data/jar-requests/medium-confidence-pcl200-pcl250.json
Comment (sam803) [reply api-tests/data/jar-requests/medium-confidence-pcl200-pcl250.json:1]: I guess I could check the `journeyType` parameter for `pcl200` and `pcl250` then set them accordingly without having to alter the regex?

Comment (Joe-Edwards-GDS) [reply api-tests/data/jar-requests/medium-confidence-pcl200-pcl250.json:1]: I think this is ok for now, especially if we only have like 3 different variants, and we can fix it up later if need be.

If you did want to change I'd probably refactor that step as:
`Given I start a new 'P2' journey`
`Given I start a new 'P2, PCL250' journey`
etc.

In future we're going to also need P1 journeys, and to distinguish 'identity proving' from 'reverification' journeys - but that's about it.

File: api-tests/src/steps/ipv-steps.ts [hunk 5]
```diff
@@ -104,7 +112,7 @@ When(
   ): Promise<void> {
     let attempt = 1;
     while (attempt <= MAX_ATTEMPTS) {
-      await startNewJourney(this, journeyType, false);
+      await startNewJourney(this, journeyType, false, undefined);
```
Comment (Joe-Edwards-GDS) [inline api-tests/src/steps/ipv-steps.ts:115]: If you make the last parameter optional (i.e. `journeyId?: string`) then you wouldn't need this - but this is nice and explicit

**Review by Joe-Edwards-GDS — APPROVED** [review]
Thanks, looks a bit more idiomatic now :)

**Pull request merged**